deps/Makefile

```diff
@@ -246,11 +246,11 @@ include $1.version
 ifneq ($(NO_GIT),1)
 $2_SRC_DIR = $1
 $1:
-	git clone -q --depth=50 --branch $$($2_BRANCH) $$($2_GIT_URL) $1
+	git clone -q --depth=50 $$($2_GIT_URL) $1
 	touch -c $1/$3
 $1/$3: $1.version | $1
-	cd $1 && git fetch -q $$($2_GIT_URL) $$($2_BRANCH):remotes/origin/$$($2_BRANCH)
-	cd $1 && git checkout -q --detach remotes/origin/$$($2_BRANCH)
+	cd $1 && git fetch -q $$($2_GIT_URL)
```

Comment:
cc @vtjnash the fact that we're doing a shallow clone may cause problems if we try to change `*.version` to a commit that's on a different branch, I was doing some dummy tests and it wasn't always working as I'd hoped.